src/shared/creds-util.c

```diff
@@ -223,7 +224,10 @@ static int make_credential_host_secret(
         assert(dfd >= 0);
         assert(fn);
 
-        fd = openat(dfd, ".", O_CLOEXEC|O_WRONLY|O_TMPFILE, 0400);
+        /* used by linkat(2) with AT_EMPTY_PATH flag to create a link to a file referred to by a file
+           descriptor */
+        if (have_effective_cap(CAP_DAC_READ_SEARCH))
```

Comment:
Why is it necessary to check for the cap? It went through the fallback path before, no?

Reply:
The fail is later, in the `linkat`.  Seems that a non-root user can still do the `openat` here, and fail later in the `linkat`.

The condition is here to decide to use the alternative path, that will later avoid the `linkat`, because `t` is set.

Reply:
got it, maybe expand the comment to say that it's intended to make non-root usage work

Reply:
Should be done